File: Python/Project/model/HeapSort.py
class HeapSort:

    def __init__(self, size):
        from model.MinHeap import MinHeap
        self.min_heap = MinHeap(size)

    def sort(self, array=[]):
        output = []
        for i in range(0, len(array)):
            self.min_heap.add(array[i])
        while not self.min_heap.is_empty():
            output.append(self.min_heap.pop())
        return output

File: C#/comparingAlgorithmsCSharp/comparingAlgorithmsCSharp/Heapsort.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace comparingAlgorithmsCSharp
{
   
        class Heapsort
    {

        private MinHeap heap;


        public Heapsort(int maxSize)
        {

            heap = new MinHeap(maxSize);

        }

        public List<int> Sort(int[] list)
        {

            List<int> sortList = new List<int>();

            foreach (int i in list)
            {

                heap.Add(i);

            }

            while (heap.IsEmpty()==false)
            {

                sortList.Add(heap.Pop());

            }

            return sortList;
        }
    }
}
